### DISCUSSION TITLE
Does JQuery works on PHP Native?

### DISCUSSION THREAD

**Discussion by JDMF19** [NativePHP/laravel#350] [Q&A]
I'm trying to use Jquery on my PHP Native app but it shows this error:
![Screenshot_77](https://github.com/user-attachments/assets/862ba769-2ed3-4a10-8d19-1cf8eb9f0714)

**Comment by callmeahmedr** [comment]
The error message indicates that the jQuery library is not defined. This means that either the jQuery library file is not included in your HTML file or there's an issue with the path or syntax of the include statement.

Make sure to have this in your website's `<head>` tags:
`<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.6.0/jquery.min.js"></script>`

If you're still having trouble, use your browser's developer tools to inspect the network tab and see if the jQuery file is being loaded successfully. You can also use the console to check for any additional error messages or warnings.

Happy coding!

**Reply by JDMF19** [reply]
ok, something strange is happening to me. 
look, when I load jquery from my project (is the same script that you just gave me I copied it and pasted on jquery-3.6.0.min.js)
![Screenshot_84](https://github.com/user-attachments/assets/467ca155-4761-4bba-9257-22ed817dea6b)
👆 here don't works 

In the console shows script is loaded 👇
![Screenshot_86](https://github.com/user-attachments/assets/df19517b-ebc3-4eca-b6f7-c8ec8aa5da64)


BUT when I include the script directly from https://ajax.googleapis.com/ajax/libs/jquery/3.6.0/jquery.min.js
and keep my jquery-3.6.0.min.js loaded too
![Screenshot_89](https://github.com/user-attachments/assets/572e077f-f2b9-499e-a02b-246bd08a6731)

IT WORKS 😱 
![Screenshot_88](https://github.com/user-attachments/assets/d22bb766-695d-476d-b714-5b6502208b07)

but only if I keep both scripts loaded it's so strange I know but thanks I'll just load both, if it works I don't care why

**Reply by callmeahmedr** [reply]
Try this: 

- Double-check the path to your local `jquery-3.6.0.min.js` file. Ensure there are no typos or incorrect directory references.
- Your browser might be caching the local jQuery file and ignoring the newer version. Try clearing your browser cache and reloading the page.

**Comment by simonhamp** [comment]
Can you show how you're trying to load jQuery into your pages?

**Reply by JDMF19** [reply]
those are my scripts and it works when I open it like normal Laravel web page but when I run command *php artisan native:serve* it don't works. Thats way I ask becase it seems strange to me.
![Screenshot_83](https://github.com/user-attachments/assets/be8221ed-926a-439a-a670-578884fdfd26)

**Comment by JDMF19** [comment]
I found the solution it seems it was detecting that was an a Node/CommonJS environment and some strange things happened to the scripts
so I put this code above all scripts and now all scripts works properly

👇
<script>if (typeof module === 'object') {window.module = module; module = undefined;}</script>
